### DISCUSSION TITLE
Can I write to a buffer in parallel using some set of compute pipelines?

### DISCUSSION THREAD

**Discussion by dalduba** [gpuweb/gpuweb#4131] [Q&A]
The issue is to build some big buffer from a sequence of other buffers (N>limit of max storage buffers per shader). The building process is not trivial, it is more complex than just concatenation. I have a shader that puts the data from some input buffer to concrete place in out buffer, that depends on the current input buffer. Thus there is no collisions in out data ranges if I run N pipelines for N input buffers.
Here was an answer about pipelines parallel behavior. https://github.com/gfx-rs/wgpu/discussions/2999

And I suggest that there is no chance to call the pipelines in parallel.

Is this right? Are there any suggestions?

**Comment by Kangz** [comment]
The answer in the issue you linked applies to WebGPU as well: two `DispatchWorkgroups` calls that have write-write or read-write dependencies on the set of resources they use are guaranteed to not run in parallel. The "big buffer" seems like it will be written by each dispatch so that's the write-write case. In the future WebGPU may gain support for more explicitly controlling the parallelism during compute but that's not the case today.
